Sources/Workspace/Workspace.swift

```diff
@@ -1718,12 +1719,23 @@ extension Workspace {
                 targetName: artifact.targetName
             ]
 
-            if let existingArtifact = existingArtifact, case .remote = existingArtifact.source {
-                // If we go from a remote to a local artifact, we can remove the old remote artifact.
-                artifactsToRemove.append(existingArtifact)
+            if artifact.path.extension == "zip" {
+                // If we already have an artifact that was extracted from an archive with the same checksum,
+                // we don't need to extract the artifact again.
+                if case .local(let existingChecksum) = existingArtifact?.source,
+                   existingChecksum == self.checksum(forBinaryArtifactAt: artifact.path, diagnostics: diagnostics) {
+                    continue
+                }
```

Comment:
formatting ^^

Reply:
`SwiftFormat` [moves the bracket to the next line](https://github.com/apple/swift-package-manager/pull/3746/commits/5e2d7bbe8937164b5b8cadfcdd15d4066c423886), unfortunately, it is very inconvenient to run it since it formats a lot of not affected by the PR code (so I have to discard a lot of not relevant changes each time), what would be the right way to check formatting?
```
swiftformat .                                                                           
Running SwiftFormat...
Reading config file at /Users/.../swift-package-manager/.swiftformat
SwiftFormat completed in 1.8s.
449/652 files formatted, 1 file skipped.
```

Reply:
yea. what I normally do is point it to a single file / directory: `swiftformat "/path/to/your/code/"`

Reply:
oh but I was reading the code wrong originally, the formatting is actually fine

Reply:
Okay, then I reverted the change. Running `swiftformat Sources/Workspace/Workspace.swift` also leads to hundreds of changes in the `Workspace` file, it looks like `swiftformat` was introduced a few years later the file was created, and it hasn't been formatted since then.